scripts/missions/asa/04_Sugar-Coated_Directive.lua

```diff
@@ -0,0 +1,277 @@
+-----------------------------------
+-- A Shantotto Ascension
+-- A Shantotto Ascension M4
+-----------------------------------
+-- !addmission 11 3
+-- Trodden Snow  : !pos -19.7 -17.3 104.4 126
+-----------------------------------
+require('scripts/globals/missions')
+require('scripts/settings/main')
+require('scripts/globals/interaction/mission')
+require('scripts/globals/zone')
+-----------------------------------
+local flamesID  = require("scripts/zones/Cloister_of_Flames/IDs")
+local frostID   = require("scripts/zones/Cloister_of_Frost/IDs")
+local galesID   = require("scripts/zones/Cloister_of_Gales/IDs")
+local stormsID  = require("scripts/zones/Cloister_of_Storms/IDs")
+local tidesID   = require("scripts/zones/Cloister_of_Tides/IDs")
+local tremorsID = require("scripts/zones/Cloister_of_Tremors/IDs")
+local qufimID   = require("scripts/zones/Qufim_Island/IDs")
+-----------------------------------
+
+local mission = Mission:new(xi.mission.log_id.ASA, xi.mission.id.asa.SUGAR_COATED_DIRECTIVE)
+
+mission.reward =
+{
+    nextMission = { xi.mission.log_id.ASA, xi.mission.id.asa.ENEMY_OF_THE_EMPIRE_I },
+}
+
+mission.sections =
+{
+    {
+        check = function(player, currentMission, missionStatus, vars)
+            return currentMission == mission.missionId
+        end,
+
+        [xi.zone.CLOISTER_OF_FLAMES] =
+        {
+            ['Fire_Protocrystal'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasKeyItem(xi.ki.DOMINAS_SCARLET_SEAL) and mission:getVar(player, 'Ifrit') == 0 then
+                        mission:setVar(player, 'Ifrit', 1)
```

Comment:
Lots of vars going around here, this could be simplified to just use a single var as a bitfield

Reply:
I am unsure on how to do this.